### PR TITLE
wav2vec2 converter: create the proper vocab.json while converting fairseq wav2vec2 finetuned model

### PR DESCRIPTION
**Pull request by cceyda**
# What does this PR do?
While converting a finetuned wav2vec2 model we also need to convert the related dictionary `dict.ltr.txt` to hugging face `vocab.json` format.

If a `dict_path` is specified:
  - Creates&saves the necessary vocab.json file  
  - Modifies config file special token ids and vocab size accordingly
  - Creates a processor with the right special tokens and saves the processor `preprocessor_config.json`

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Did you make sure to update the documentation with your changes? -> not sure if there are any docs related to this.
- [ ] Did you write any new necessary tests? -> not sure if there are tests related to this.

## Who can review?

Models:
- wav2vec2: @patrickvonplaten @LysandreJik

### CODE REVIEW

**Comment by LysandreJik** [comment]
Thanks for opening a PR @cceyda! I'll let Patrick review this one, but he's off for a couple of weeks. 

Maybe @patil-suraj can give it a look in the meantime, but we'll probably have to wait for Patrick to review as he has the most extensive experience with w2v2!

File: src/transformers/models/wav2vec2/convert_wav2vec2_original_pytorch_checkpoint_to_pytorch.py
Comment (Getmany1) [inline src/transformers/models/wav2vec2/convert_wav2vec2_original_pytorch_checkpoint_to_pytorch.py:201]: The value for `return_attention_mask` depends on whether a model has been pre-trained using attention mask. For models with `feat_extract_norm='group'` it should be False and for `feat_extract_norm='layer'` True:

```
feature_extractor = Wav2Vec2FeatureExtractor(
                feature_size=1, sampling_rate=16000, padding_value=0, do_normalize=True, return_attention_mask=True if config.feat_extract_norm=='layer' else False
            )
```

Comment (cceyda) [reply src/transformers/models/wav2vec2/convert_wav2vec2_original_pytorch_checkpoint_to_pytorch.py]: I didn't know that, fixed it now~ Thank you

**Review by patil-suraj — APPROVED** [review]
LGTM! Thanks a lot for adding this!

**Comment by patil-suraj** [comment]
@LysandreJik  Merging this PR. The changes look good to me and do not affect the modeling file at all, so it won't cause any issues there.

**Comment by LysandreJik** [comment]
Works for me! Thanks @cceyda and @patil-suraj

**Pull request merged**
